apis/elasticache/v1beta2/zz_replicationgroup_types.go

```diff
@@ -396,13 +397,20 @@ type ReplicationGroupParameters struct {
 	AtRestEncryptionEnabled *bool `json:"atRestEncryptionEnabled,omitempty" tf:"at_rest_encryption_enabled,omitempty"`
 
 	// Password used to access a password protected server. Can be specified only if transit_encryption_enabled = true.
+	// If you set autoGenerateAuthToken to true, the Secret referenced here will be created or updated with generated auth token if it does not already contain one.
 	// +kubebuilder:validation:Optional
 	AuthTokenSecretRef *v1.SecretKeySelector `json:"authTokenSecretRef,omitempty" tf:"-"`
 
 	// Strategy to use when updating the auth_token. Valid values are SET, ROTATE, and DELETE. Defaults to ROTATE.
 	// +kubebuilder:validation:Optional
 	AuthTokenUpdateStrategy *string `json:"authTokenUpdateStrategy,omitempty" tf:"auth_token_update_strategy,omitempty"`
 
+	// Password used to access a password protected server. Can be specified only if transit_encryption_enabled = true.
+	// If true, the auth token will be auto-generated and stored in the Secret referenced by the authTokenSecretRef field.
+	// +upjet:crd:field:TFTag=-
+	// +kubebuilder:validation:Optional
+	AutoGenerateAuthToken *bool `json:"autoGenerateAuthToken,omitempty" tf:"-"`
```

Comment:
nit: In theory, we could also support auto token generation for the `spec.initProvider` API for completeness. But we do not need to do this in this PR as I don't expect it to be a real use case.

Reply:
👍  I think we should do the same every place common.PasswordGenerator is used then (in another PR)